### PR TITLE
Fixes accountsesdt index

### PR DESCRIPTION
**Pull request by miiu96**
- Added  the `shardID` field  in the `accounts` and `accountsesdt` index
- Delete data from `accountsesdt` and `accountsesdthistory` in case of rollback

### CODE REVIEW

File: process/elasticProcessor.go [hunk 1]
```diff
@@ -313,7 +319,27 @@ func (ei *elasticProcessor) RemoveTransactions(header coreData.HeaderHandler, bo
...
-	return ei.elasticClient.DoBulkRemove(elasticIndexer.TransactionsIndex, encodedTxsHashes)
+	return ei.elasticClient.DoQueryRemove(
+		elasticIndexer.TransactionsIndex,
+		converters.PrepareHashesForQueryRemove(encodedTxsHashes),
+	)
+}
+
+// RemoveAccountsESDT will remove data from accountsesdt index and accountsesdthistory
+func (ei *elasticProcessor) RemoveAccountsESDT(headerTimestamp uint64) error {
+	query := fmt.Sprintf(`{"query": {"bool": {"must": [{"match": {"shardID": {"query": %d,"operator": "AND"}}},{"match": {"timestamp": {"query": "%d","operator": "AND"}}}]}}}`, ei.selfShardID, headerTimestamp)
```
Comment (bogdan-rosianu) [inline process/elasticProcessor.go:330]: there is still a `"query": %d` is it ok?

File: client/elasticClient.go [hunk 2]
```diff
@@ -15,8 +15,6 @@ import (
 	"github.com/elastic/go-elasticsearch/v7/esapi"
 )
 
-// TODO add more unit tests
```
[the remark this answers is not in the document]
Comment (miiu96) [reply client/elasticClient.go:18]: added again and made a task in order to do some refactoring in the `esclient`

File: client/elasticClient.go [hunk 3]
```diff
@@ -152,46 +150,54 @@ func (ec *elasticClient) DoMultiGet(ids []string, index string, withSource bool,
...
 	}
 
-	res, err := ec.es.DeleteByQuery(
+	res, err := ec.client.DeleteByQuery(
 		[]string{index},
-		&body,
-		ec.es.DeleteByQuery.WithIgnoreUnavailable(true),
+		body,
+		ec.client.DeleteByQuery.WithIgnoreUnavailable(true),
+		ec.client.DeleteByQuery.WithConflicts("proceed"),
```
[the remark this answers is not in the document]
Comment (miiu96) [reply client/elasticClient.go:163]: added

File: converters/json.go [hunk 4]
```diff
@@ -24,3 +26,16 @@ func JsonEscape(i string) string {
...
 	return string(b[1 : len(b)-1])
 }
+
+// PrepareHashesForQueryRemove will prepare the provided hashes for query remove
+func PrepareHashesForQueryRemove(hashes []string) *bytes.Buffer {
+	if len(hashes) == 0 {
+		hashes = []string{}
+	}
+
+	hashesM, _ := json.Marshal(hashes)
```
[the remark this answers is not in the document]
Comment (miiu96) [reply converters/json.go:36]: done.

File: process/elasticProcessor.go (hunk 1 above)
Comment (miiu96) [reply process/elasticProcessor.go:331]: yes is ok.

File: process/accounts/accountsProcessor.go [hunk 5]
```diff
@@ -20,12 +20,13 @@ import (
 
 var log = logger.GetOrCreate("indexer/process/accounts")
 
-// accountsProcessor a is structure responsible for processing accounts
+// accountsProcessor is structure responsible for processing accounts
```
Comment (sstanculeanu) [inline process/accounts/accountsProcessor.go:23]:
```suggestion
// accountsProcessor is a structure responsible for processing accounts
```

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**